golem-cli/tests/api_deployment_fileserver.rs

```diff
@@ -116,16 +116,32 @@ fn api_deployment_file_server_simple(
         &host,
     ])?;
 
+    // read a read-only file
     let res1 = reqwest::blocking::get(format!("http://{host}/files/{component_id}/foo.txt"))?;
     assert_eq!(res1.status().as_u16(), 200);
     assert_eq!(res1.headers().get("content-type").unwrap(), "text/plain");
     assert_eq!(res1.text()?, "foo\n");
 
+    // read a read-write file
     let res2 = reqwest::blocking::get(format!("http://{host}/files/{component_id}/bar.txt"))?;
     assert_eq!(res2.status().as_u16(), 200);
     assert_eq!(res2.headers().get("content-type").unwrap(), "text/plain");
     assert_eq!(res2.text()?, "bar\n");
 
+    // read a non-existing file
+    let res3 = reqwest::blocking::get(format!("http://{host}/files/{component_id}/notfound.txt"))?;
+    assert_eq!(res3.status().as_u16(), 404);
+
+    // try to read a directory. This will fail with a 400 as the mime type of 'dir' cannot be inferred.
+    let res4 = reqwest::blocking::get(format!("http://{host}/files/{component_id}/dir"))?;
+    assert_eq!(res4.status().as_u16(), 400);
+
+    // TODO: read a file from a nested path
```

Comment:
Leaving this one for a separate pr as it requires some more work. Let's discuss tomorrow